### PR TITLE
Implement MPSVector

### PR DESCRIPTION
**Pull request by tgymnich**
before:
```julia
julia> @btime A * B;
  23.166 μs (60 allocations: 1.49 KiB)

julia> @btime A * b;
  161.667 μs (327 allocations: 7.23 KiB)
```
after:
```julia
julia> @btime A * b;
  15.333 μs (47 allocations: 1.11 KiB)
```

### CODE REVIEW

File: lib/mps/linalg.jl [hunk 1]
```diff
@@ -92,6 +92,60 @@ function LinearAlgebra.herk_wrapper!(C::MtlMatrix, tA::AbstractChar, A::MtlMatri
 end
 end
 
+const MPS_VALID_MATVECMUL_TYPES =
```
Comment (maleadt) [inline lib/mps/linalg.jl:95]: Wasn't Int8 also listed in the CI message a while ago?

Comment (tgymnich) [reply lib/mps/linalg.jl:95]: True, but I could never get it to work. And Apple also does not document the supported types for this kernel for whatever reason.

File: test/mps.jl [hunk 2]
```diff
@@ -35,6 +35,33 @@ if MPS.is_supported(current_device())
...
+            arr_b = rand(input_jl_type, (rows))
+            arr_c = zeros(accum_jl_type, (rows))
+
+            buf_a = MtlArray{input_jl_type}(arr_a)
+            buf_b = MtlArray{input_jl_type}(arr_b)
+            buf_c = MtlArray{accum_jl_type}(undef, (rows))
+
+            truth_c = (alpha .* accum_jl_type.(arr_a)) *  accum_jl_type.(arr_b) .+ (beta .* arr_c)
+
+            Metal.@sync MPS.matvecmul!(buf_c, buf_a, buf_b, alpha, beta)
```
[the remark this answers is not in the document]
Comment (tgymnich) [reply test/mps.jl:58]: Test is copied from mat mul. I guess I'll remove that too then...

File: lib/mps/linalg.jl [hunk 3]
```diff
@@ -92,6 +92,60 @@ function LinearAlgebra.herk_wrapper!(C::MtlMatrix, tA::AbstractChar, A::MtlMatri
...
+    [(Float16, Float16),
+     (Float16, Float32),
+     (Float32, Float32)]
+
+function LinearAlgebra.generic_matvecmul!(C::MtlVector, tA::AbstractChar, A::MtlMatrix, B::MtlVector, _add::MulAddMul)
+    if ndims(A) > 2
+        throw(ArgumentError("A has more than 2 dimensions"))
+    end
```
Comment (maleadt) [inline lib/mps/linalg.jl:103]: I don't think MtlMatrix can ever have ndims != 2?

**Review by maleadt — COMMENTED** [review]
LGTM

File: lib/mps/linalg.jl (hunk 3 above)
Comment (tgymnich) [reply lib/mps/linalg.jl]: fair

**Comment by maleadt** [comment]
Awesome, thanks!

**Pull request merged**